lib/smack/SmackRep.cpp

```diff
@@ -1162,6 +1172,68 @@ std::string SmackRep::getPrelude() {
   }
   s << "\n";
 
+  for (size_t s1 = 0; s1 < INTEGER_SIZES.size(); ++s1) {
```

Comment:
I think this block should be moved into the `Prelude` class, maybe in a function called `generatePrelude`. That way prelude generation would be more encapsulated. Unless you have a good reason for putting it here.

Reply:
I put it there because `SmackRep.cpp` already had a `getPrelude()` function, and I thought it would make more sense to put all prelude generation code in a single function rather than separating it into two different functions across different classes. Then again, creating the `Prelude.cpp` class results in this effect anyway. Do you think `SmackRep.cpp` should have a `getPrelude()` function if we're creating a dedicated `Prelude` class, or do you think it's fine if some of the prelude generation happens in `SmackRep.cpp`?

Reply:
I see. I think that ideally all of the prelude generation should be moved into your `Prelude` class. I hope that there are not too many dependencies that have to be passed to it from `SmackRep`. I just looked at the code and it doesn't seem that bad. So `SmackRep` should not have `getPrelude` once this is done.